kubernetes-client/src/main/java/io/micronaut/kubernetes/client/graalvm/KubernetesClientFeature.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.graalvm;
+
+
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import io.micronaut.core.annotation.Internal;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Stream;
+
+/**
+ * Configures native image generation.
+ *
+ * @author Pavol Gressa
+ * @since 3.1
+ */
+@AutomaticFeature
+@Internal
+public class KubernetesClientFeature implements Feature {
```

Comment:
It's not necessary to do it but we have this utility class in core https://github.com/micronaut-projects/micronaut-core/blob/3.0.x/core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java

There are helper methods to make it easy to deal with the GraalSDK API. We use them for example in micronaut-sql: https://github.com/micronaut-projects/micronaut-sql/blob/master/jdbc/src/main/java/io/micronaut/jdbc/nativeimage/JdbcFeature.java